src/librustc_mir_build/hair/pattern/_match.rs

```diff
@@ -1664,7 +1669,9 @@ crate fn is_useful<'p, 'tcx>(
                     bug!("Encountered or-pat in `v` during exhaustiveness checking")
                 }
             }
-            matrix.push(v);
+            if !is_under_guard {
```

Comment:
Maybe add a comment here as well re. the purpose of this check.

Reply:
4b1aa9fa